File: py_fluent/leetCode/zuoye/quickSort.cpp
#include<iostream>
#include<algorithm>
#include<vector>
using namespace std;

template<class T>
void quickSorted(T nums[],int left,int right){
    T temp=nums[left];
    int low=left;
    int height=right;
    if(left>=right)return;
    while (left<right){
        while(nums[right]>=temp && left<right){
            --right;
            break;
        }
         nums[left]=nums[right];
        while(nums[left]<=temp && left<right){

            ++left;
            break;
        }
         nums[right]=nums[left];
    }
    nums[left]=temp;
    quickSorted(nums,low,left);
    quickSorted(nums,left+1,height);
}

int main(){
    char nums[]="asdjhkiuqowe";
    quickSorted(nums,0,11);
    for_each(nums,nums+11,[](auto &x){cout<<x; });
}

File: programerTest/test1/quickSort.py
def quickSort(nums,Left,Right):
      left=Left;
      right=Right;
      if left>=right:
          return ;
      temp=nums[left];
      while left<right:
          while left<right and nums[right]>=temp:
              right-=1;
          nums[left]=nums[right];
          while left<right and nums[left]<temp:
              left+=1;
          nums[right]=nums[left];
      nums[left]=temp;
      quickSort(nums,Left,left-1);
      quickSort(nums,right+1,Right);

